oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -1018,7 +1018,8 @@ public void getRequestMetadata(
 
     // If scopes are provided but we cannot use self signed JWT, then use scopes to get access
     // token.
-    if (!createScopedRequired() && !useJwtAccessWithScope) {
+    if (!createScopedRequired() && !useJwtAccessWithScope
+        || serviceAccountUser != null && serviceAccountUser.length() > 0) {
```

Comment:
somehow I think the logic would be
```
// We don't use self signed JWT approach if `useJwtAccessWithScope` is false, or domain wide
// delegation is used.
if (!useJwtAccessWithScope || (serviceAccountUser != null && serviceAccountUser.length() > 0)) {
    return super.getRequestMetadata(uri);
}
```
Most likely I wrote this code, but I don't remember why `!createScopedRequired()` is here, lol